### DISCUSSION TITLE
How to parse JWT Header fields?

### DISCUSSION THREAD

**Discussion by schuerg** [lestrrat-go/jwx#405] [Q&A] [answered]
It seems that the `jwt.Parse()` function only decodes the JWT payload but not the JWT header fields like `alg`, `typ` and `kid`.

So how to parse a given JWT in compact base64 form and get the header fields? Espacially the `kid` value?

**Comment by lestrrat** [comment, marked as answer]
@schuerg I'm going to take a wild guess, but I think you are confusing JWT and JWS.  JWT may arrive to you in signed format, but at that point it's a _JWS_ message. Header fields like `alg` and `kid` only exist in the JWS message. So if you want to manually do something with the JWS headers, you should be looking at [`jws.Parse`](https://pkg.go.dev/github.com/lestrrat-go/jwx@v1.2.1/jws#Parse)
